docs/source/building/variants.rst

```diff
@@ -0,0 +1,676 @@
+information that went into the hash is stored with the package, in a file at
+``info/hash_input.json``. Currently, only the first 7 characters of the hash are
+stored. Output package names will keep the pyXY and npXYY for now, but have
+added the 7-character hash. Your package names will look like:
+
+``my-package-1.0-py27h3142afe_0.tar.bz2``
+
+Since conflicts only need to be prevented within one version of a package, we
+think this will be adequate. If you run into hash collisions with this limited
+subspace, please file an issue on the conda-build issue tracker.
+
+The information that goes into this hash is currently defined in conda-build's
+metadata.py module; the _get_hash_contents member function. This function
+captures the following information:
+
+* ``source`` section
+* ``requirements`` section
+* ``build`` section, except:
+  * ``number``
+  * ``string``
+* any other recipe files, such as bld.bat, build.sh, etc. Every file other than
+  meta.yaml is part of the hash.
+
+All "falsey" values (e.g. empty list values) are removed.
+
+There is a CLI tool that just pretty-prints this json file for easy viewing:
+
+.. code-block:: shell
+
+   conda inspect hash-inputs <package path>
+
+This produces output such as:
+
+.. code-block:: shell
+
+   {'test_rm_rf_does_not_follow_links-1.0-h7330_0': {u'build': {u'script': u'python setup.py install --single-version-externally-managed --record=record.txt'},
+                                                  u'requirements': {u'build': [u'openssl 1.0.2k 0',
+                                                                               u'pip 9.0.1 py27_1',
+                                                                               u'python 2.7.13 0',
+                                                                               u'readline 6.2 2',
+                                                                               u'setuptools 27.2.0 py27_0',
+                                                                               u'sqlite 3.13.0 1',
+                                                                               u'tk 8.5.18 0',
+                                                                               u'wheel 0.29.0 py27_0',
+                                                                               u'zlib 1.2.8 3']},
+                                                  u'source': {u'path': u'/Users/msarahan/code/conda-build/tests/test-recipes/split-packages/_rm_rf_stays_within_prefix'}}}
+
+
+.. _extra_jinja2:
+
+Extra Jinja2 functions
+----------------------
+
+
+Two especially common operations when dealing with these API and ABI
+incompatibilities are ways of specifying such compatibility, and of explicitly
+expressing the compiler to be used. Three new Jinja2 functions are available when
+evaluating ``meta.yaml`` templates:
+
+* ``pin_compatible``: To be used as pin in run and/or test requirements. Takes
```

Comment:
Another nitpick: An easier to read format for arguments might be:
pin_compatible(package)